Sources/Transaction/Inputs/MixinSelectionStrategy.swift

```diff
@@ -26,3 +26,56 @@ extension MixinSelectionStrategy {
             ringSize: McConstants.RING_SIZE)
     }
 }
+
+protocol IdempotentMixinSelectionStragegy: MixinSelectionStrategy {
+    func selectMixinIndices(
+        rng: MobileCoinRng,
+        forRealTxOutIndices realTxOutIndices: [UInt64],
+        selectionRange: PartialRangeUpTo<UInt64>?,
+        excludedTxOutIndices: [UInt64],
+        ringSize: Int
+    ) -> [Set<UInt64>]
+}
+
+extension IdempotentMixinSelectionStragegy {
+    func selectMixinIndices(
+        rng: MobileCoinRng,
+        forRealTxOutIndices realTxOutIndices: [UInt64],
+        selectionRange: PartialRangeUpTo<UInt64>?,
+        excludedTxOutIndices: [UInt64] = []
+    ) -> [Set<UInt64>] {
+        selectMixinIndices(
+            rng: rng,
+            forRealTxOutIndices: realTxOutIndices,
+            selectionRange: selectionRange,
+            excludedTxOutIndices: excludedTxOutIndices,
+            ringSize: McConstants.RING_SIZE)
+    }
+
+    func selectMixinIndices(
+        forRealTxOutIndices realTxOutIndices: [UInt64],
+        selectionRange: PartialRangeUpTo<UInt64>?,
+        excludedTxOutIndices: [UInt64] = []
+    ) -> [Set<UInt64>] {
+        selectMixinIndices(
+            rng: MobileCoinDefaultRng(),
```

Comment:
feel like the `IdempotentMixinSelectionStragegy` should use the chacha20 RNG by default

Reply:
nevermind, I understand how this works now